src/main/java/Freedom.java

```diff
@@ -0,0 +1,256 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import freedom.exceptions.CannotCreateDirectory;
+import freedom.exceptions.CannotReadFile;
+import freedom.exceptions.InvalidCommand;
+import freedom.tasks.Task;
+import freedom.tasks.ToDo;
+import freedom.tasks.Deadline;
+import freedom.tasks.Event;
+
+public class Freedom {
+    protected static ArrayList<Task> tasks = new ArrayList<>();
+    static int lastIndex = 0;
+
+    static final String LOGO = "\t________________________________________\n";
+    static final String DATA_FILE_PATH = "./data/freedom.txt";
+
+    public static void main(String[] args) {
+        final String START_MESSAGE = """
+                \tHello! I'm Freedom
+                \tWhat can I do for you?
+                """;
+        final String CLOSING_MESSAGE = "\tBye. Hope to see you again soon!\n";
+
+        System.out.println(LOGO + START_MESSAGE + LOGO);
+        Scanner in = new Scanner(System.in);
+
+        try {
+            checkData();
+            loadData();
+
+            // Super loop for getting inputs
+            while (in.hasNextLine()) {
+                String line = in.nextLine();
+
+                if (line.equals("bye")) {
+                    break;
+                }
+                handleInput(line);
+            }
+
+            saveData();
+        } catch (Exception e) {
+            System.out.print("");
+        }
+
+        System.out.println(LOGO + CLOSING_MESSAGE + LOGO);
+    }
+
+    public static void handleInput(String input) {
```

Comment:
Methods like handleInput and editTask are quite large and handle too much responsibility. Consider splitting them into smaller methods.